native_client/javascript/index.d.ts

```diff
@@ -0,0 +1,196 @@
+/**
```

Comment:
@piraka9011 Would it make sense that this gets ingested and documented on ReadTheDocs ? If so, do you know how ? Would `jsdoc` liek we have be enough?

Reply:
I added the docs since I wasn't sure how you would package the type declaration file.
The docs you have already for `jsdoc` should be enough.
Those using type script would be able to infer the types from the declaration file instead.

Reply:
>  Those using type script would be able to infer the types from the declaration file instead.

Right, but can we make that simpler to read / discover somehow ? More than mentionning in the JS doc that "here we have the types" ? I don't know what's the usual thing to do in the typescript community.

Reply:
Docstrings in the JS code only are fine IMO.
There's no 'standard' with TS that I'm aware of (Was there ever one for web dev tools 🙃)
Example is [`@types/node`](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/node/v8) or `octokit/rest.js`.
Node has the official docs but typescript specific bindings for something like `Buffer` for example are documented. Octokit on the other hand has most of their docs and types as part of their `octokit/types.ts` repo which wrap around some other libraries and JS modules.

If you have `jsdoc` setup then that's sufficient for documentation. Typescript devs will immediately realize there's no type declaration module when compiling with `tsc` (like myself) if they try to import DeepSpeech.

Reply:
> Docstrings in the JS code only are fine IMO.
> There's no 'standard' with TS that I'm aware of (Was there ever one for web dev tools upside_down_face) so take my suggestions with a grain of salt. I think what's more important is consistency.
> 
> Example is [`@types/node`](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/node/v8) or `octokit/rest.js`.
> Node has the official docs but typescript specific bindings for something like `Buffer` for example are documented. Octokit on the other hand has most of their docs and types as part of their `octokit/types.ts` repo which wrap around some other libraries and JS modules.
> 
> Typescript devs will realize there's no type declaration module when compiling with `tsc` (like myself) if they try to import DeepSpeech. A simple example should be enough to get folks up and running.

So I guess including `index.d.ts` in what is parsed by `jsdoc` will be enough?

Reply:
Yup